NExtends/Collections/ObjectPropertiesList.cs

```diff
@@ -0,0 +1,362 @@
+				IndexableProperties = typeof(K).GetProperties().Where(p => p.PropertyType == typeof(T)).Select(p => p.Name).ToList();
+			}
+			else
+			{
+				IndexableProperties = Attributes.OrderBy(a => a.Order).SelectMany(a => a.IndexableProperties).ToList();
+			}
+
+			foreach (var pName in IndexableProperties)
+			{
+				var pInfo = typeof(K).GetProperty(pName);
+
+				if (pInfo == null)
+					throw new Exception(string.Format("Class {0} doesn't have a property named {1}", typeof(K).Name, pName));
+
+				var getter = pInfo.GetValueGetter<K, T>();
+				var setter = pInfo.GetValueSetter<K, T>();
+
+				getters.Add(getter);
+				setters.Add(setter);
+				propertyNames.Add(pName);
+			}
+
+			Getters = getters.ToArray();
+			Setters = setters.ToArray();
+			PropertyNames = propertyNames.ToArray();
+		}
+
+		public ObjectPropertiesList(K objectToIndex)
+		{
+			this.ObjectToIndex = objectToIndex;
+		}
+
+		public T this[int key]
+		{
+			get
+			{
+				return Getters[key](ObjectToIndex);
+			}
+			set
+			{
+				Setters[key](ObjectToIndex, value);
+			}
+		}
+
+		public void Add(T item)
+		{
+			throw new NotImplementedException();
+		}
+
+		public void Clear()
+		{
+			throw new NotImplementedException();
+		}
+
+		public bool Contains(T item)
+		{
+			bool contains = false;
+			for (int i = 0; i < Count && !contains; i++)
+				contains = this[i].Equals(item);
+
+			return contains;
+		}
+
+		public void CopyTo(T[] array, int arrayIndex)
+		{
+			if (object.ReferenceEquals(array, null))
+			{
+				throw new ArgumentNullException(
+					"Null array reference",
+					"array"
+					);
+			}
+
+			if (arrayIndex < 0)
+			{
+				throw new ArgumentOutOfRangeException(
+					"Index is out of range",
+					"index"
+					);
+			}
+
+			if (array.Rank > 1)
+			{
+				throw new ArgumentException(
+					"Array is multi-dimensional",
+					"array"
+					);
+			}
+
+			foreach (T o in this)
+			{
+				array.SetValue(o, arrayIndex);
+				arrayIndex++;
+			}
+		}
+
+		public int Count
+		{
+			get { return IndexableProperties.Count(); }
+		}
+
+		public bool IsReadOnly
+		{
+			get { return false; }
+		}
+
+		public bool Remove(T item)
+		{
+			throw new NotImplementedException();
+		}
+
+		public IEnumerator<T> GetEnumerator()
+		{
+			return new ObjectFieldArrayEnumerator<K, T>(this);
+		}
+
+		System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator()
+		{
+			return this.GetEnumerator();
+		}
+
+		public int IndexOf(T item)
+		{
+			throw new NotImplementedException();
+		}
+
+		public void Insert(int index, T item)
+		{
+			throw new NotImplementedException();
+		}
+
+		public void RemoveAt(int index)
+		{
+			throw new NotImplementedException();
+		}
+
+		public static string PropertyNameOfIndex(int idx)
+		{
+			return PropertyNames[idx];
+		}
+	}
+
+	public class ObjectFieldArrayEnumerator<K, T> : IEnumerator<T>
+	{
+		ObjectPropertiesList<K, T> Of;
+		int CurIndex = -1;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove the 'CurIndex' field and declare it as a local variable in the relevant methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=csharpsquid%3AS1450)